### PR TITLE
Update CAPV to CAPI v1alpha2

### LINKED ISSUE #511: Refactor types/dependencies/project to v1alpha2
/kind feature

This issue tracks refactoring the types, dependencies, and general project assets to CAPI v1alpha2.

/assign @akutz

### PR DESCRIPTION
**Pull request by akutz**
**What this PR does / why we need it**:
This patch is the initial commit for updating CAPV to CAPI v1alpha2. This is incomplete and will be rebased, so please keep that in mind when handling merge conflicts. Outstanding items:

- [x] Update the `./pkg/cloud/vsphere/services/govmomi` directory to use refactored types
- [x] ~Port the actuator logic to the new infrastructure controllers~ (moved to #510)
- [x] Update the CI and hack scripts with the new package locations and names

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #511, #514, #515, #516

**Special notes for your reviewer**:
This is not testable. The v1a2 update will leave `master` broken for a bit. For a complete list of issues related to the migration to v1a2, please see #472.

**Release note**:

```release-note
* Updated CAPV to CAPI v1alpha2
```

### CODE REVIEW

File: pkg/cloud/vsphere/context/machine_context.go [hunk 1]
```diff
@@ -138,19 +97,7 @@ func (c *MachineContext) String() string {
...
-// the machine. A nil value is returned if the MachineRef is not yet set.
-func (c *MachineContext) GetMoRef() *types.ManagedObjectReference {
-	if c.MachineConfig.MachineRef == "" {
-		return nil
-	}
-	return &types.ManagedObjectReference{
-		Type:  "VirtualMachine",
-		Value: c.MachineConfig.MachineRef,
-	}
+	return fmt.Sprintf("%s/%s", c.Cluster.Namespace, c.Cluster.Name, c.Machine.Name)
```
Comment (zdyxry) [inline pkg/cloud/vsphere/context/machine_context.go]: "%s/%s/%s" ?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	defaultSyncPeriod    = 10 * time.Minute
+	defaultRequeuePeriod = 20 * time.Second
+
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = infrav1.AddToScheme(scheme)
```
Comment (zdyxry) [inline main.go]: Need to add `_ = clusterv1.AddToScheme(scheme)` ? when i didn't add this ,i got error : `no kind is registered for the type v1alpha2.Cluster in scheme`

**Comment by akutz** [comment]
/assign @figo
/assign @frapposelli 

Please lgtm this when you're able to do so. This PR sets up `master` to be the basis of CAPV v1a2. There is still quite a bit of work to do, but this is the starting point.

**Comment by frapposelli** [comment]
Good stuff @akutz !

/lgtm
/hold

cancel when you see fit

**Comment by akutz** [comment]
Thanks @frapposelli! In order to facilitate more work on v1a2 by others, I'm cancelling the hold so people can begin to fix bugs and iterate towards v1a2.

/hold cancel

**Comment by frapposelli** [comment]
Awesome, thanks @akutz!

**Pull request merged**